lib/pbench/test/unit/server/database/test_datasets_metadata_db.py

```diff
@@ -211,3 +257,23 @@ def test_delete_with_metadata(self, db_session, create_user):
         # to the deleted Dataset
         metadata = Database.db_session.query(Metadata).filter_by(dataset_ref=id).first()
         assert metadata is None
+
+    def test_setgetvalue_user(self, db_session, create_user, create_drb_user):
+        ds = Dataset.create(owner=create_user.username, controller="frodo", name="fio")
+        Metadata.setvalue(dataset=ds, key="user.contact", value="Barney")
+        Metadata.setvalue(
+            dataset=ds, key="user.contact", value="Fred", user=create_user
+        )
+        Metadata.setvalue(
+            dataset=ds, key="user.contact", value="Wilma", user=create_drb_user
+        )
+
+        assert Metadata.getvalue(dataset=ds, user=None, key="user") == {
+            "contact": "Barney"
+        }
```

Comment:
This "works" only because we're testing the primitives, here, right?  Normally, if `user` were `None` execution would never reach the `setvalue()` call, and so there would never be a value there for the `getvalue()` call to retrieve, right?  (It _might_ be worth a comment to that effect, in here somewhere.)

Reply:
Yes, these are unit tests working below the level of user authentication; but that's kinda the point of unit tests. Testing of this interface has nothing to do with the semantics of higher-level interfaces, and do we really want a convention of documenting where they differ? Does it matter?